inst/testthat/helper_autotest.R

```diff
@@ -497,29 +508,89 @@ run_experiment = function(task, learner, seed = NULL, configure_learner = NULL)
   }
 
-  # predict_newdata_fast method
-  if (startsWith(task$id, "feat_all") && exists("predict_newdata_fast", learner) && learner$predict_type == "prob") {
+  # check predict_newdata_fast
+  stage = "predict_newdata_fast()"
 
+  if (startsWith(task$id, "feat_all") && exists("predict_newdata_fast", learner) && learner$predict_type == "prob") {
     prediction = suppressWarnings(try(learner$predict_newdata_fast(task$data()), silent = TRUE))
 
     if (inherits(prediction, "try-error")) {
       ok = prediction
       prediction = NULL
-      return(err(c("predict_newdata_fast failed: ", as.character(ok))))
+      return(err(error_as_character(ok)))
     }
 
     msg = checkmate::check_list(prediction)
     if (!isTRUE(msg)) {
-      return(err("predict_newdata_fast does not return a list"))
+      return(err("does not return a list"))
     }
 
     msg = checkmate::check_names(names(prediction), subset.of = mlr3::mlr_reflections$learner_predict_types[[learner$task_type]][[learner$predict_type]])
     if (!isTRUE(msg)) {
-      return(err("Names of list returned by predict_newdata_fast do not match learner predict_types: %s", str_collapse(names(prediction))))
+      return(err("Names of returned list do not match learner predict_types: %s", str_collapse(names(prediction))))
+    }
+  }
+
+  # check marshalling
+  stage = "marshal"
+
+  if (startsWith(task$id, "feat_all") && "marshal" %in% learner$properties) {
+    learner$marshal()
+    # learner$marshaled checks if model is of class "marshaled"
+    if (!learner$marshaled) {
+      return(err("model not marshaled"))
+    }
+
+    learner$unmarshal()
+    # checks if the "marshaled" class is removed
+    if (learner$marshaled) {
+      return(err("model not unmarshaled"))
+    }
+
+    prediction_marshaling = suppressWarnings(try(learner$predict(task), silent = TRUE))
+    if (inherits(prediction_marshaling, "try-error")) {
+      ok = prediction_marshaling
+      prediction_marshaling = NULL
+      return(err(error_as_character(ok)))
+    }
+  }
+
+  # check encapsulation
+  stage = "encapsulation"
+  if (startsWith(task$id, "feat_all")) {
+    learner_encapsulated = learner$clone(deep = TRUE)
+    learner_encapsulated$encapsulate("callr", default_fallback(learner_encapsulated))
+
+    # check learner state because encapsulation catches errors
+    learner_encapsulated$train(task)
+    log = learner_encapsulated$log[stage == "train"]
+    if ("error" %in% log$class) {
+      return(err("train log has errors: %s", mlr3misc::str_collapse(log[class == "error", msg])))
+    }
+
+    learner_encapsulated$predict(task)
+    log = learner_encapsulated$log[stage == "predict"]
+    if ("error" %in% log$class) {
+      return(err("predict log has errors: %s", mlr3misc::str_collapse(log[class == "error", msg])))
+    }
+
+    # while resampling, less information are stored in the state which can lead to errors
+    rr = resample(task, learner_encapsulated, rsmp("holdout"), store_models = TRUE)
```

Comment:
Maybe we should only run the `resample()` test, what's the additional benefit of running `$train()` and `$predict()` manually?

Reply:
Done